src/main/java/com/sleekbyte/tailor/output/ViolationSuppressor.java

```diff
@@ -0,0 +1,71 @@
+package com.sleekbyte.tailor.output;
+
+import com.sleekbyte.tailor.common.Location;
+import com.sleekbyte.tailor.common.Messages;
+import com.sleekbyte.tailor.listeners.CommentAnalyzer;
+import org.antlr.v4.runtime.Token;
+
+import java.util.List;
+import java.util.Stack;
+
+/**
+ * Ignore analysis on lines that are marked "tailor:disable".
+ */
+public final class ViolationSuppressor extends CommentAnalyzer {
+    private static final String DISABLE_LINE_PATTERN = "// tailor:disable";
+    private static final String TAILOR_OFF = "// tailor:off";
+    private static final String TAILOR_ON = "// tailor:on";
+
+    /**
+     * Create instance of ViolationSuppressor.
+     *
+     * @param printer            An instance of Printer
+     * @param singleLineComments List of // comments
+     * @param multilineComments  List of /* comments
+     */
+    public ViolationSuppressor(Printer printer, List<Token> singleLineComments, List<Token> multilineComments) {
+        super(printer, singleLineComments, multilineComments);
+    }
+
+    @Override
+    public void analyze() {
+        Stack<Integer> ignoreBlockBeginStack = new Stack<>();
+        Token lastSuppressViolationComment = null;
```

Comment:
Since `Token` is not a primitive, we need to explicitly initialize it to `null`.